backend/src/apiserver/config/config_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2025 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/list"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/model"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/resource"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func fakeResourceManager() *resource.ResourceManager {
+	clientManager := resource.NewFakeClientManagerOrFatalV2()
+	resourceManager := resource.NewResourceManager(clientManager, &resource.ResourceManagerOptions{CollectMetrics: false})
+	return resourceManager
+}
+
+func TestLoadSamplesConfigBackwardsCompatibility(t *testing.T) {
+	rm := fakeResourceManager()
+	pc := []deprecatedConfig{
+		{
+			Name:        "Pipeline 1",
+			Description: "test description",
+			File:        "testdata/sample_pipeline.yaml",
+		},
+		{
+			Name:        "Pipeline 2",
+			Description: "test description",
+			File:        "testdata/sample_pipeline.yaml",
+		},
+	}
+
+	path, err := writeSampleConfigDeprecated(t, pc, "sample.json")
+	require.NoError(t, err)
+	err = LoadSamples(rm, path)
+	require.NoError(t, err)
+
+	_, err = rm.GetPipelineByNameAndNamespace(pc[0].Name, "")
+	require.NoError(t, err)
+	_, err = rm.GetPipelineByNameAndNamespace(pc[1].Name, "")
+	require.NoError(t, err)
+
+	_, err = rm.GetPipelineVersionByName(pc[0].Name)
+	require.NoError(t, err)
+	_, err = rm.GetPipelineVersionByName(pc[1].Name)
+	require.NoError(t, err)
+
+	// Update pipeline version for Pipeline 1
+	pc[0].Name = "Pipeline 3"
+	path, err = writeSampleConfigDeprecated(t, pc, "sample.json")
+	require.NoError(t, err)
+
+	// Loading samples should result in no pipeline uploaded
+	err = LoadSamples(rm, path)
+	require.NoError(t, err)
+	_, err = rm.GetPipelineByNameAndNamespace(pc[0].Name, "")
+	var userErr *util.UserError
+	if assert.ErrorAs(t, err, &userErr) {
+		require.Equal(t, codes.NotFound, userErr.ExternalStatusCode())
+	}
+}
+
+func TestLoadSamples(t *testing.T) {
+	rm := fakeResourceManager()
+	pc := config{
+		LoadSamplesOnRestart: true,
+		Pipelines: []configPipelines{
+			{
+				Name:               "Pipeline 1",
```

Comment:
Could you add a test case for multiple pipeline versions for the same pipeline?

Reply:
I might not be following, there is a case for updating `Pipeline 1 - Ver 1` -> `Pipeline 1 - Ver 2`, did you mean to add more cases updating it further?

Reply:
The existing test always only ever has a config with one pipeline version per pipeline.

I was suggesting a test case where you have multiple pipeline versions for the same pipeline in the `configPipelines` slice/array.